test/core/shared-everything-threads/shared-everything-threads/globals.wast

```diff
@@ -0,0 +1,429 @@
+;; Check the `shared` attribute on globals.
+
+(module
+  ;; Imported.
+  (global (import "spectest" "global_i32") (shared i32))
+  (global (import "spectest" "global_i32_mut") (shared mut i32))
+  (global (import "spectest" "global_i64") (shared i64))
+  (global (import "spectest" "global_i64_mut") (shared mut i64))
+  (global (import "spectest" "global_f32") (shared i32))
+  (global (import "spectest" "global_f32_mut") (shared mut f32))
+  (global (import "spectest" "global_f64") (shared f64))
+  (global (import "spectest" "global_f64_mut") (shared mut f64))
+  (global (import "spectest" "global_v128") (shared v128))
+  (global (import "spectest" "global_v128_mut") (shared mut v128))
+
+  ;; Initialized.
```

Comment:
You could also test that shared global initializers can only reference other shared globals, but that unshared global initializers can use both shared and unshared globals.

Reply:
Yes, exactly. And then again where `$a` is shared and there is no error.